python/ray/_raylet.pyx

```diff
@@ -1818,6 +1822,28 @@ cdef class CoreWorker:
         with nogil:
             CCoreWorkerProcess.RunTaskExecutionLoop()
 
+    def exit_worker(self, exit_type: str, c_string detail):
```

Comment:
This is needed to exit the worker with a proper error messages. Without this, the worker will restart again and again with spammy error messages to the driver (and after 30 seconds, it will fail). By exposing this API, we can kill the worker with a proper error messages.